src/main/java/io/pravega/perf/WriterWorker.java

```diff
@@ -101,9 +101,16 @@ public void benchmark() throws InterruptedException, IOException {
                 final String header = String.format(TIME_HEADER_FORMAT, System.currentTimeMillis());
                 final String data = buffer.replace(0, TIME_HEADER_SIZE, header).toString();
                 writeData(data);
+                /*
+                flush is required here for following reasons:
+                1. The writeData is called for End to End latency mode; hence make sure that data is sent.
+                2. In case of kafka benchmarking, the buffering makes too many writes;
+                   flushing moderates the kafka producer.
+                3. If the flush called after several iterations, then flush may take too much of time.
+                */
+                flush();
```

Comment:
So, this means that we are flushing for both Pravega and Kafka in the same way, right? This is only needed if you want to measure end-to-end latency, so if you execute a write-only workload you are not going to flush for every event? 
Another question: does it make sense that the invocation of `flush()` after every write could be configurable? For instance, in `fio` benchmark you can use `buffered` or  `direct` IO. Similarly, you may want to execute a benchmark with the buffering the system uses by default and flushing after every write as a config parameter in the benchmark.

Reply:
yes.. its in same way now.. 
for write only.. it does not flush every event.

for flushing for every write is possible , but, confirm if its required so that i will create another PR for implementing the same.

Reply:
In write-only mode, when the client is flushing? Apparently, it looks that after it writes all the events, but maybe I'm wrong. If this is the case, this may need to be configurable (e.g., after how many writes the writer flushes). This may be interesting to analyze the impact of flushing on latency and throughput.

Reply:
Hi Raul, In write only mode also, now I have implemented "-flush" option in which you can supply the number of events per producer to flush. I have tested with both kafka and pravega.